### PR TITLE
[IMP] hr_employee: improve form usability

### PR DESCRIPTION
**Pull request by SimonGenin**
Improve slightly the layout of some forms related to employee and contracts.

Modified modules:
hr
hr_expense
hr_holidays
timesheet_grid
hr_contract_salary
hr_payroll_account

Task ID 2277298

### CODE REVIEW

File: addons/hr/views/hr_employee_views.xml [hunk 1]
```diff
@@ -105,9 +105,10 @@
                                             <field name="address_id"
                                                 context="{'show_address': 1}"
                                                 options='{"always_reload": True, "highlight_first_line": True}'/>
+                                            <field name="work_location"/>
                                         </group>
-                                        <group name="managers" string="Responsibles">
-                                            <field name="coach_id"/>
+                                        <group name="managers" string="Approvers">
```
Comment (jbm-odoo) [inline addons/hr/views/hr_employee_views.xml:110]: If you have only HR module installed, you will have a title "Approvers" with nothing bellow....

File: addons/hr/views/hr_employee_views.xml [hunk 2]
```diff
@@ -105,9 +105,10 @@
...
+                                            <!-- used to contain the coach id field. Keep the group for overrides in other modules -->
```
Comment (jbm-odoo) [inline addons/hr/views/hr_employee_views.xml:111]: Comments should generally provide information on the current code. You can put: `is overidded in other modules` or something like that

File: addons/hr/views/hr_employee_views.xml (hunk 1 above)
Comment (SimonGenin) [reply addons/hr/views/hr_employee_views.xml:110]: Not really sure what's the best option here. 
If I remove the group, all the inheritance is broken. 
If I remove the string, the label won't be here. 
If I have to update all the children this piece of code, it becomes cumbersome and  really not code friendly. 

Is there maybe a way to hide if it contains no element ?

Comment (jbm-odoo) [reply addons/hr/views/hr_employee_views.xml:110]: They aren't way to hide if it contains no element. 
But you can add attribute invisible 
```suggestion
                                        <group name="managers" string="Approvers" invisible="1">
```

And when you add something in this group, you add also  
```
            <xpath expr="//group[@name='managers']" position="attributes">
                <attribute name="invisible">0</attribute>
            </xpath>
```

Comment (SimonGenin) [reply addons/hr/views/hr_employee_views.xml]: I've just pushed this solution. I am really not found of it as it is too much for such tiny details.

File: addons/hr/views/hr_employee_views.xml [hunk 3]
```diff
@@ -89,12 +89,12 @@
                                 <field name="mobile_phone" widget="phone"/>
                                 <field name="work_phone" widget="phone"/>
                                 <field name="work_email" widget="email"/>
-                                <field name="work_location"/>
                                 <field name="company_id" groups="base.group_multi_company"/>
                             </group>
                             <group>
                                 <field name="department_id"/>
                                 <field name="parent_id"/>
+                                <field name="coach_id"/>
```
Comment (jbm-odoo) [inline addons/hr/views/hr_employee_views.xml:97]: You moved the coach_id in hr.employee view. Why not make this change in hr.employee.public and res.users view to keep consistency between these views?

Comment (SimonGenin) [reply addons/hr/views/hr_employee_views.xml:97]: I didn't know those views existed. But I can't make the change.. This invisible attribute is awful, many files need an update and every futur person wishing to add a field there will have to think to add the invisble = 0 attribute override. This is the opposite of best practice. 
I'd rather have an empty approvers group title once in a while than the proposed solution.
I revert the changes made yesterday. I'll ask the PO if it is that important to not have the coach in the approvers

Comment (SimonGenin) [reply addons/hr/views/hr_employee_views.xml:97]: @jbm-odoo So I just had a call with the PO. coach_id must leave this.
So, any idea how we could hide it without having to "unhide" it in the many overrides ? 
Another solution would also to get rid of it completely and recreate it in all the override. Same amount of work, but less hacky. But I don't like it either.

Comment (jbm-odoo) [reply addons/hr/views/hr_employee_views.xml:97]: I gave you my best solution. If you really think re-creating the group in all the override is prettier and the same amount of work, I'll let you do it. But in any case, it's not ok to have an empty group shown.

File: addons/hr_holidays/views/hr_views.xml [hunk 4]
```diff
@@ -118,6 +118,9 @@
             <xpath expr="//group[@name='managers']" position="inside">
                 <field name="leave_manager_id"/>
             </xpath>
+            <xpath expr="//group[@name='managers']" position="attributes">
+                <attribute name="invisible">0</attribute>
+            </xpath>
```
Comment (jbm-odoo) [inline addons/hr_holidays/views/hr_views.xml:123]: No longer needed

File: addons/hr_holidays/views/hr_views.xml [hunk 5]
```diff
@@ -215,6 +218,9 @@
             <xpath expr="//group[@name='managers']" position="inside">
                 <field name="leave_manager_id" attrs="{'readonly': [('can_edit', '=', False)]}"/>
             </xpath>
+            <xpath expr="//group[@name='managers']" position="attributes">
+                <attribute name="invisible">0</attribute>
+            </xpath>
```
Comment (jbm-odoo) [inline addons/hr_holidays/views/hr_views.xml:223]: No longer needed

**Comment by SimonGenin** [comment]
@jbm-odoo All done

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**